src/pages/search/advanced_search/PublisherFilter.tsx

```diff
@@ -0,0 +1,90 @@
+import { Autocomplete, Typography } from '@mui/material';
+import { useQuery } from '@tanstack/react-query';
+import { useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { useHistory } from 'react-router';
+import { searchForPublishers } from '../../../api/publicationChannelApi';
+import { ResultParam } from '../../../api/searchApi';
+import { AutocompleteTextField } from '../../../components/AutocompleteTextField';
+import { Publisher } from '../../../types/registration.types';
+import { useDebounce } from '../../../utils/hooks/useDebounce';
+
+export const PublisherFilter = () => {
+  const { t } = useTranslation();
+  const history = useHistory();
+  const searchParams = new URLSearchParams(history.location.search);
+  const publisherParam = searchParams.get(ResultParam.Publisher);
+  const [publisherQuery, setPublisherQuery] = useState('');
+  const debouncedQuery = useDebounce(publisherQuery);
+
+  const publisherOptionsQuery = useQuery({
+    queryKey: ['publisherSearch', debouncedQuery],
+    enabled: debouncedQuery.length > 3 && debouncedQuery === publisherQuery,
+    queryFn: () => searchForPublishers(debouncedQuery, '2023'),
+    meta: { errorMessage: t('feedback.error.get_publishers') },
+  });
+
+  const publisherList = publisherOptionsQuery.data?.hits ?? [];
+
+  const selectedPublisherQuery = useQuery({
+    enabled: !!publisherParam,
+    queryKey: [publisherParam],
+    queryFn: () => searchForPublishers(publisherParam ?? '', '2023'),
+    meta: { errorMessage: t('feedback.error.get_publisher') },
+    staleTime: Infinity,
+  });
+
+  useEffect(() => {
+    if (selectedPublisherQuery.data) {
+      setPublisherQuery(selectedPublisherQuery.data.hits[0].name);
+    }
+  }, [selectedPublisherQuery.data]);
+
+  const handleChange = (selectedValue: Publisher | null) => {
+    if (selectedValue) {
+      searchParams.set(ResultParam.Publisher, selectedValue.identifier);
+    } else {
+      searchParams.delete(ResultParam.Publisher);
+    }
+
+    history.push({ search: searchParams.toString() });
+  };
+
+  return (
+    <Autocomplete
+      sx={{ minWidth: '15rem' }}
+      value={publisherParam && selectedPublisherQuery.data?.hits[0] ? selectedPublisherQuery.data.hits[0] : null}
+      isOptionEqualToValue={(option, value) => option.id === value.id}
+      options={
+        debouncedQuery && publisherQuery === debouncedQuery && !publisherOptionsQuery.isLoading ? publisherList : []
+      }
+      filterOptions={(options) => options}
+      inputValue={publisherQuery}
+      onInputChange={(_, newInputValue) => {
+        setPublisherQuery(newInputValue);
+      }}
+      onChange={(_, newValue) => {
+        handleChange(newValue);
+      }}
+      blurOnSelect
+      disableClearable={!publisherQuery}
+      loading={publisherOptionsQuery.isFetching}
```

Comment:
Når man har en param i URL vil denne basere seg på feil query nå, så forelår å lage en egen const for å håndtere loading: 
`const isLoading = publisherParam ? selectedPublisherQuery.isLoading : publisherOptionsQuery.isLoading;`
```suggestion
      loading={isLoading}
```
Sammen endring blir da også på `AutocompleteTextField`